### ISSUE TITLE
'' Currently `save` requires model to be a graph network. '' when i try to save model using model.save

### ISSUE THREAD

**Issue by zlian2016** [keras-team/keras#12892]
An error occur: '' keras Currently `save` requires model to be a graph network. '' when I try to save model using model.save(). Here is my code.

class Encoder(tf.keras.Model):
    def __init__(self):
        super(Encoder, self).__init__()
        self.fc = tf.keras.layers.Dense(units=EMBEDDING_DIM, input_shape=(1, 4096))

    def call(self, inputs, training=None, mask=None):
        return self.fc(inputs)

... ...
... ...

encoder = Encoder()

optimizer = tf.train.AdamOptimizer()
variables = encoder.variables + embedding.variables + decoder.variables
gradients = tape.gradient(batch_loss, variables)
optimizer.apply_gradients(zip(gradients, variables), tf.train.get_or_create_global_step())

encoder.save(filepath='./encoder.h5', overwrite=False, include_optimizer=True)

I need some help! Thanks in advance!

**Comment by jvishnuvardhan** [comment]
Please fill the issue template [here](https://github.com/keras-team/keras/issues/new/choose). Could you update them if they are relevant in your case, or leave them as N/A? Along with the template, please provide as many details as possible to find the root cause of the issue. It would be great if you can provide a small code to reproduce the error. Thanks!

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by maggiemh** [comment]
hello, I met the same problem, have you solved this problem? if it has, could you please share your solution?
